drivers/power_domain/power_domain_gpio.c

```diff
@@ -109,17 +109,17 @@ static int pd_gpio_init(const struct device *dev)
 	return rc;
 }
 
-#define POWER_DOMAIN_DEVICE(id)						   \
-	static const struct pd_gpio_config pd_gpio_##id##_cfg = {	   \
-		.enable = GPIO_DT_SPEC_INST_GET(id, enable_gpios),	   \
-		.startup_delay_us = DT_INST_PROP(id, startup_delay_us),	   \
-		.off_on_delay_us = DT_INST_PROP(id, off_on_delay_us),	   \
-	};								   \
-	static struct pd_gpio_data pd_gpio_##id##_data;			   \
-	PM_DEVICE_DT_INST_DEFINE(id, pd_gpio_pm_action);		   \
-	DEVICE_DT_INST_DEFINE(id, pd_gpio_init, PM_DEVICE_DT_INST_GET(id), \
-			      &pd_gpio_##id##_data, &pd_gpio_##id##_cfg,   \
-			      POST_KERNEL, 75,				   \
+#define POWER_DOMAIN_DEVICE(id)						    \
+	static const struct pd_gpio_config pd_gpio_##id##_cfg = {	    \
+		.enable = GPIO_DT_SPEC_INST_GET(id, enable_gpios),	    \
+		.startup_delay_us = DT_INST_PROP(id, startup_delay_us),	    \
+		.off_on_delay_us = DT_INST_PROP(id, off_on_delay_us),	    \
+	};								    \
+	static struct pd_gpio_data pd_gpio_##id##_data;			    \
+	PM_DEVICE_DT_INST_DEFINE(id, pd_gpio_pm_action);		    \
+	DEVICE_DT_INST_DEFINE(id, pd_gpio_init, PM_DEVICE_DT_INST_GET(id),  \
+			      &pd_gpio_##id##_data, &pd_gpio_##id##_cfg,    \
+			      POST_KERNEL, DT_INST_PROP(id, init_priority), \
```

Comment:
this makes me think it's time to review and remove/re-think the init priorities, they're a pain and we have DT ords now... cc: @mbolivar-nordic

Reply:
#22545 is the issue tracking that feature. I have done some work/thinking recently about how to achieve that.

Reply:
@gmarull please take another look